### ISSUE TITLE
[FEAT] Pixel-precision animation

### ISSUE THREAD

**Issue by davidbrochart** [TomJGooding/textual-slider#40]
### Is there an existing issue for this?

- [x] I have searched the open issues

### Description of the feature or enhancement

To get pixel-precision animation, maybe take inspiration from the [Switch](https://textual.textualize.io/widget_gallery/#switch) widget?

https://github.com/TomJGooding/textual-slider/assets/4711805/9ca4d83b-9e35-4ded-b4bd-6e8d08aaa791

**Comment by TomJGooding** [comment]
Thanks @davidbrochart  for your interest and suggestion! 

Can I just check you don't mean pixel-precision in terms of moving the slider with the mouse (where the mouse position is based on character cells rather than pixels)? Rather this enhancement would just make the transition when moving the slider (either with the mouse or keyboard) a bit more smooth by adding a small animation?

**Comment by davidbrochart** [comment]
You're right, at first I wanted pixel-precision slider control but I realized it's not possible because mouse coordinates only have character-precision.
Pixel-precision animation would still be nice, but only if you think it's worth it and not too difficult.
Anyway, awesome project, keep up the great work!

**Comment by TomJGooding** [comment]
I'm a bit torn about animating the slider. While the effect is nice with a longer animation duration, I think for a slider it should be more snappy. But then the effect is so subtle I wonder if it is worth it?

https://github.com/user-attachments/assets/daa58155-8daa-4537-a203-dd1118162db2

(Thanks David for your patience, this project has been on the back burner for a bit!)

**Comment by davidbrochart** [comment]
> (Thanks David for your patience, this project has been on the back burner for a bit!)

No worries, and thanks for working on it!
Would it make sense to have the animation duration configurable?

**Comment by TomJGooding** [comment]
> Would it make sense to have the animation duration configurable?

Maybe. Out of interest, would you opt for the shorter or longer animation?

**Comment by davidbrochart** [comment]
The 0.3s duration looks good to me.

**Comment by TomJGooding** [comment]
There are several ways to update the slider position:

1. Programmatically by setting the slider value
2. Using the left and right arrow keys
3. Clicking and dragging the slider thumb
4. Clicking to quickly jump to a specific position (feature added in [#46](https://github.com/TomJGooding/textual-slider/pull/46) but not yet released)

All these slider position updates could _potentially_ be animated for a smoother transition. But should all of them be animated? Moreover, not all animations should have the same duration. For example, dragging the thumb may feel sluggish if it shares the same animation duration as when the arrow keys are pressed.

I might need to give this some more thought.

**Comment by davidbrochart** [comment]
Right, thanks for looking into all the consequences of that feature.
Maybe we could take inspiration from Textual's scollbar?

**Comment by TomJGooding** [comment]
I'm doing a spring clean in my GitHub - sorry but I don't think animations are really necessary.
